text_localization_environment/TextLocEnv.py

```diff
@@ -70,12 +83,31 @@ def step(self, action):
             reward = np.sign(new_iou - self.iou)
             self.iou = new_iou
 
-        self.history.insert(0, self.to_one_hot(action))
-        self.history.pop()
+        return reward
 
-        self.state = self.compute_state()
+    def calculate_potential_reward(self, action):
+        old_bbox = self.bbox
+        old_iou = self.iou
+
+        if self.action_set[action] != self.trigger:
+            self.action_set[action]()
+
+        reward = self.calculate_reward(action)
+
+        self.bbox = old_bbox
+        self.iou = old_iou
+
+        return reward
+
+    def find_positive_actions(self):
+        rewards = np.array([self.calculate_potential_reward(i) for i in self.action_set])
+
+        positive_actions = np.arange(0, self.action_space.n)[rewards > 0]
+
+        if len(positive_actions) == 0:
+            return np.arange(0, self.action_space.n)
```

Comment:
Here you return a `ndarray`, whereas in line 110 you return a `list`. You might choose one type for both of the return statements.